Network/Socket/Types.hsc

```diff
@@ -906,6 +907,10 @@ data SockAddr
         String           -- sun_path
   deriving (Eq, Ord, Typeable)
 
+instance NFData SockAddr where
+  rnf (SockAddrUnix str) = rnf str
+  rnf _ = ()
```

Comment:
Could you enumerate all of the other constructors instead of using a wildcard. That leads to a more maintainable instance over the long run.